src/platform/monitoring/frontend-metrics/metrics.js

```diff
@@ -0,0 +1,166 @@
+  if (typeof paintEntries === 'undefined') {
+    return false;
+  }
+  return paintEntries[0].startTime;
+}
+
+/**
+ * Parse a PerformanceNavigationTiming object into a metrics payload [FormData] object to send to backend.
+ * Also includes first-contentful-paint metric if it's available.
+ *
+ * @param {PerformanceNavigationTiming} An object with information from the browser's Performance API
+ * @returns {FormData} A new FormData object with a data parameter, as a JSON string of metrics data
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/FormData
+ */
+// Exported for unit tests
+export function buildMetricsPayload(entry) {
+  const totalPageLoad = entry.duration;
+  const firstByte = entry.responseStart;
+  const domProcessing = entry.domComplete - entry.domInteractive;
+  const domComplete = entry.domComplete - entry.requestStart;
+  const domInteractive = entry.domInteractive;
+
+  let metrics = {
+    totalPageLoad,
+    firstByte,
+    domProcessing,
+    domComplete,
+    domInteractive,
+  };
+
+  if (contentfulPaintEntry()) {
+    const firstContentfulPaint = contentfulPaintEntry();
+    metrics = Object.assign({
+      firstContentfulPaint
+    }, metrics);
+  }
+  const metricsPayload = new FormData();
+  const metricsArray = [];
+  const pageUrl = window.location.pathname;
+
+  Object.keys(metrics).forEach(metric => {
+    metricsArray.push({
+      metric,
+      duration: metrics[metric]
+    });
+  });
+
+  // eslint-disable-next-line camelcase
+  const data = JSON.stringify({ metrics: metricsArray, page_id: pageUrl });
+  metricsPayload.append('data', data);
+
+  return metricsPayload;
+}
+
+/**
+ * Sends a payload via beacon. Asynchronously transmits data to the web server
+ * when the User-Agent has an opportunity to do so, without delaying the unload event
+ * or affecting the performance of the next navigation.
+ *
+ * @returns {boolean}
+ * @see: https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon
+ */
+// Exported for unit tests
+export function sendMetricsToBackend(metricsPayload) {
+  const url = `${environment.API_URL}/v0/performance_monitorings`;
+  if (!navigator.sendBeacon(url, metricsPayload)) {
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Capture browser-provided navigation timing metrics from a PerformanceObserver object.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/PerformanceObserver/observe
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming
+ */
+// Exported for unit tests
+export function captureMetrics() {
+  const observer = new PerformanceObserver(list => {
+    list.getEntriesByType('navigation').forEach(entry => {
+      const metricsPayload = buildMetricsPayload(entry);
+      window.addEventListener('unload', () => {
+        sendMetricsToBackend(metricsPayload);
+      });
+    });
+  });
+
```

Comment:
I think this was commented by @jbalboni before, but I've had weird experiences with the `unload` event, so I'm wondering if it would be more reliable to invert this by having only one listener for the `unload` event that iterates through all of the `entry` items, instead of creating a listener for each item.

Reply:
Just making sure my suggestion didn't break anything here? Not sure if `getEntriesByType` needed to be outside of that `unload` event or not lol.